account_invoice_supplier_ref_unique/models/account_invoice.py

```diff
@@ -0,0 +1,70 @@
+# Copyright 2016 Acsone
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import _, api, fields, models
+from odoo.exceptions import ValidationError
+
+
+class AccountInvoice(models.Model):
+    _inherit = "account.invoice"
+
+    supplier_invoice_number = fields.Char(
+        string='Vendor invoice number',
+        readonly=True,
+        states={'draft': [('readonly', False)]},
+        copy=False)
+
+    @api.one
+    @api.constrains('supplier_invoice_number')
+    def _check_unique_supplier_invoice_number_insensitive(self):
+        """
+        Check if an other vendor bill has the same supplier_invoice_number
+        and the same commercial_partner_id than the current instance
+        """
+        if self.supplier_invoice_number and\
+                self.type in ('in_invoice', 'in_refund'):
+            same_supplier_inv_num = self.search([
+                ('commercial_partner_id', '=', self.commercial_partner_id.id),
+                ('type', 'in', ('in_invoice', 'in_refund')),
+                ('supplier_invoice_number',
+                 '=ilike',
+                 self.supplier_invoice_number),
+                ('id', '!=', self.id), ])
```

Comment:
`limit=1` in order to avoid use index [0] in the line 38 `same_supplier_inv_num[0].supplier_invoice_number`

Reply:
Yes indeed! Thank you.